packages/trpc/server/routers/loggedInViewer/updateProfile.handler.ts

```diff
@@ -257,24 +256,11 @@ export const updateProfileHandler = async ({ ctx, input }: UpdateProfileOptions)
   }
 
   if (user.timeZone !== data.timeZone && updatedUser.schedules.length > 0) {
-    // on timezone change update timezone of default schedule
-    const defaultScheduleId = await getDefaultScheduleId(user.id, prisma);
-
-    if (!user.defaultScheduleId) {
-      // set default schedule if not already set
-      await prisma.user.update({
-        where: {
-          id: user.id,
-        },
-        data: {
-          defaultScheduleId,
-        },
-      });
-    }
-
     await prisma.schedule.updateMany({
       where: {
-        id: defaultScheduleId,
+        id: {
+          in: input.availabilityIds,
+        },
```

Comment:
This is a very dangerous change. We don't validate availabilityIds are present since this field is optional. This will trigger an update of ALL schedules. Already affected production.

Reply:
cc @keithwillcode @emrysal @sean-brydon

Reply:
Have to be honest. This seems like bad default behavior by prisma. Will reach out to them for insights

Reply:
My mistake 😥😫! Will keep it in mind next time

Reply:
Opened https://github.com/calcom/cal.com/pull/15721 to prevent this from happening in the future